python/hail/linalg/blockmatrix.py

```diff
@@ -77,47 +154,44 @@ def from_numpy(cls, ndarray, block_size=None):
         if not block_size:
             block_size = BlockMatrix.default_block_size()
 
-        # convert to float64 to ensure that jarray method below works
-        ndarray = ndarray.astype(np.float64)
         if ndarray.ndim != 2:
             raise FatalError("from_numpy: ndarray must have two axes, found shape {}".format(ndarray.shape))
         n_rows, n_cols = ndarray.shape[0:2]
         if n_rows == 0 or n_cols == 0:
             raise FatalError("from_numpy: ndarray dimensions must be non-zero, found shape {}".format(ndarray.shape))
+        if ndarray.dtype != np.float64:
+            ndarray = ndarray.astype(np.float64)
 
-        # np.ravel() exports row major by default
-        data = jarray(Env.jvm().double, np.ravel(ndarray))
-        # breeze is column major
-        is_transpose = True
-
-        bdm = Env.hail().utils.richUtils.RichDenseMatrixDouble.apply(n_rows, n_cols, data, is_transpose)
-        hc = Env.hc()
+        local_temp_dir = new_local_temp_dir()
+        path = local_temp_dir + '/binary'
+        uri = local_path_uri(path)
 
-        return cls(Env.hail().linalg.BlockMatrix.fromBreezeMatrix(hc._jsc, bdm, block_size))
+        ndarray.tofile(path)
+        return cls.fromfile(uri, n_rows, n_cols, block_size)
 
     @classmethod
     @typecheck_method(entry_expr=expr_numeric,
                       path=nullable(str),
                       block_size=nullable(int))
     def from_entry_expr(cls, entry_expr, path=None, block_size=None):
-        """Create a block matrix from a numeric matrix table entry expression.
+        """Creates a block matrix from a numeric matrix table entry expression.
 
         Notes
         -----
         For shuffle resiliency, this functions writes the resulting block matrix
         to disk and then reads the result. By specifying a path, this block
         matrix can be read again using :meth:`BlockMatrix.read`. Otherwise, a
```

Comment:
Again, too much detail, although perhaps we can promise it is shuffle free (although I could imagine we'd change that if we had a reliable shuffle strategy).